src/multisig/signing_protocol.cpp

```diff
@@ -514,8 +580,35 @@ static bool compute_keys_for_destinations(
       unique_std_recipients.insert(dst_entr.addr);
   }
 
-  if (not reconstruction) {
-    tx_secret_key = rct::rct2sk(rct::skGen());
+  // figure out how many tx secret keys are needed
+  // - tx aux keys: add if there are > 1 non-change recipients, with at least one to a subaddress
+  const std::size_t num_destinations = destinations.size();
+  const bool need_tx_aux_keys = unique_subbaddr_recipients.size() + bool(unique_std_recipients.size()) > 1;
+
+  const std::size_t num_tx_keys = 1 + (need_tx_aux_keys ? num_destinations : 0);
+
+  // make tx secret keys
+  std::vector<crypto::secret_key> all_tx_secret_keys;
+  make_tx_secret_keys(tx_secret_key_seed, num_tx_keys, all_tx_secret_keys);
+
+  // split up tx secret keys
+  crypto::secret_key tx_secret_key_temp;
+  std::vector<crypto::secret_key> tx_aux_secret_keys_temp;
+  if (not collect_tx_secret_keys(all_tx_secret_keys, tx_secret_key_temp, tx_aux_secret_keys_temp))
+    return false;
+
+  if (reconstruction)
```

Comment:
This if case shouldn't need to exist. We can not communicate the keys and simply always generate them here. That may not be a change worth it, as it's a minor optimization which changes data availability as this code is flowed through (and removes a sanity check while doing so), yet something to consider.

Reply:
The keys have to be transmitted in the `pending_tx` structure. My first attempt at this just sent the entropy, but then I had test failures because there is a `wallet2::sanity_check()` that needs the tx secret keys.

Reply:
ACK